libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2207,59 +2107,51 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 		m_errorReporter.typeError(
 			5511_error,
 			arguments.front()->location(),
-			"Expected first argument to be a function pointer, not \"" +
-			type(*arguments.front())->humanReadableName() +
-			"\"."
+			"Expected first argument to be a function pointer, not \"" + type(*arguments.front())->humanReadableName()
+				+ "\"."
 		);
 		return;
 	}
 
-	if (
-		externalFunctionType->kind() != FunctionType::Kind::External &&
-		externalFunctionType->kind() != FunctionType::Kind::Declaration
-	)
+	if (externalFunctionType->kind() != FunctionType::Kind::External
+		&& externalFunctionType->kind() != FunctionType::Kind::Declaration)
 	{
 		std::string msg = "Expected regular external function type, or external view on public function.";
 
 		switch (externalFunctionType->kind())
 		{
-			case FunctionType::Kind::Internal:
-				msg += " Provided internal function.";
-				break;
-			case FunctionType::Kind::DelegateCall:
-				msg += " Cannot use library functions for abi.encodeCall.";
-				break;
-			case FunctionType::Kind::Creation:
-				msg += " Provided creation function.";
-				break;
-			case FunctionType::Kind::Event:
-				msg += " Cannot use events for abi.encodeCall.";
-				break;
-			case FunctionType::Kind::Error:
-				msg += " Cannot use errors for abi.encodeCall.";
-				break;
-			default:
-				msg += " Cannot use special function.";
+		case FunctionType::Kind::Internal:
+			msg += " Provided internal function.";
+			break;
```

Comment:
I think that no one on the team is really a fan of our current `switch` nesting style so we could change it if the formatter allows for it.